idaes/models_extra/power_generation/unit_models/cross_flow_heat_exchanger_1D.py

```diff
@@ -0,0 +1,1041 @@
+                (0-D control volumes only). Copying will generally be faster, but inlet states may not contain
+                all properties required elsewhere.
+            duty: initial guess for heat duty to assist with initialization. Can be a Pyomo expression with units.
+
+        Returns:
+            Pyomo solver results object.
+
+        """
+        # Set solver options
+        init_log = idaeslog.getInitLogger(
+            model.name, self.get_output_level(), tag="unit"
+        )
+        solve_log = idaeslog.getSolveLogger(
+            model.name, self.get_output_level(), tag="unit"
+        )
+
+        solver_obj = get_solver(self.config.solver, self.config.solver_options)
+
+        hot_side = model.hot_side
+        cold_side = model.cold_side
+        t0 = model.flowsheet().time.first()
+        if not (
+            "temperature" in hot_side.properties[t0, 0].define_state_vars().keys()
+            and "temperature" in cold_side.properties[t0, 0].define_state_vars().keys()
+        ):
+            raise NotImplementedError(
+                "Presently, initialization of the CrossFlowHeatExchanger1D requires "
+                "temperature to be a state variable of both hot side and cold side "
+                "property packages. Extension to enth_mol or enth_mass as state variables "
+                "is straightforward---feel free to open a pull request implementing it."
+            )
+
+        hot_units = model.config.hot_side.property_package.get_metadata().derived_units
+        cold_units = (
+            model.config.cold_side.property_package.get_metadata().derived_units
+        )
+
+        if model.config.shell_is_hot:
+            shell = model.hot_side
+            tube = model.cold_side
+            shell_has_pressure_change = model.config.hot_side.has_pressure_change
+            tube_has_pressure_change = model.config.cold_side.has_pressure_change
+            shell_units = (
+                model.config.hot_side.property_package.get_metadata().derived_units
+            )
+            tube_units = (
+                model.config.cold_side.property_package.get_metadata().derived_units
+            )
+        else:
+            shell = model.cold_side
+            tube = model.hot_side
+            shell_has_pressure_change = model.config.cold_side.has_pressure_change
+            tube_has_pressure_change = model.config.hot_side.has_pressure_change
+            shell_units = (
+                model.config.cold_side.property_package.get_metadata().derived_units
+            )
+            tube_units = (
+                model.config.hot_side.property_package.get_metadata().derived_units
+            )
+
+        # Trigger creation of cp for use in future initialization
+        # Important to do before initializing property packages in
+        # case it is implemented as Var-Constraint pair instead of
+        # an Expression
+        value(hot_side.properties[t0, 0].cp_mol)
```

Comment:
I took those out, but then Pylint complained the statements were pointless, so I put them back in.

Reply:
Went back and took the values out while disabling Pylint warnings for them because there's no guarantee the evaluation won't cause some error before the model is initialized.